From d8acfbb332c6aefac6a28b67e8723a83c4659bde Mon Sep 17 00:00:00 2001
From: juntingwang <Junting.Wang@windriver.com>
Date: Fri, 1 Mar 2013 16:02:56 +0800
Subject: [PATCH 04/18] Set PS2 Keyboard controller as not a ketboard type

The Eventhub detects the platforms built-in PS/2 keyboard controller
as a physical keyboard. So the system will not display the software
keyboard by default. Modify it device type in Keyboard-less tablets,
so that ActivityManager will not ignore the PS/2 controller.

Issue: AXIA-1875

Change-Id: Ia5b7592e82aaf1970066bab5e9ca5f46ac83ab1c
Signed-off-by: juntingwang <Junting.Wang@windriver.com>
---
 services/inputflinger/EventHub.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/services/inputflinger/EventHub.cpp b/services/inputflinger/EventHub.cpp
index f315ceb77b..1816082a51 100644
--- a/services/inputflinger/EventHub.cpp
+++ b/services/inputflinger/EventHub.cpp
@@ -1328,7 +1328,10 @@ status_t EventHub::openDeviceLocked(const char *devicePath, bool ignoreAlreadyOp
 
         // 'Q' key support = cheap test of whether this is an alpha-capable kbd
         if (hasKeycodeLocked(device, AKEYCODE_Q)) {
-            device->classes |= INPUT_DEVICE_CLASS_ALPHAKEY;
+            if ((device->identifier.name != "AT Translated Set 2 keyboard") ||
+                    !property_get_bool("ro.ignore_atkbd", 0)) {
+                device->classes |= INPUT_DEVICE_CLASS_ALPHAKEY;
+            }
         }
 
         // See if this device has a DPAD.
-- 
2.17.1

